pandas/core/window/rolling.py

```diff
@@ -1889,6 +1898,13 @@ def _validate(self):
             else:
                 self._win_freq_i8 = freq.nanos
 
+            if self._on.dtype == "M8[us]":
+                self._win_freq_i8 /= 1e3
+            elif self._on.dtype == "M8[ms]":
+                self._win_freq_i8 /= 1e6
+            elif self._on.dtype == "M8[s]":
+                self._win_freq_i8 /= 1e9
```

Comment:
So the problem here is that we are checking with hardcoded "M8[s]" types, and so this `if` check fails if you have tz-aware data:

```
(Pdb) self._on.dtype
datetime64[s, UTC]
(Pdb) self._on.dtype == "M8[us]"
False
```

So we should generalize this check. I think we can do something like

```suggestion
            else:
                self._win_freq_i8 = freq.nanos
                unit = dtype_to_unit(self._on.dtype)
                if unit == "us":
                    self._win_freq_i8 /= 1e3
                elif unit == "ms":
                    self._win_freq_i8 /= 1e6
                elif unit == "s":
                    self._win_freq_i8 /= 1e9
```

Because we know here that we have a `datetime64` dtype here (numpy or pd.DatetimeTzDtype), this should work. This `dtype_to_unit` function is currently defined in `pandas.core.arrays.datetimelike` (we should maybe move that, but that can be done later, for now fine to import it from there)